hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/DeleteContainerCommandHandler.java

```diff
@@ -67,6 +67,9 @@ public void handle(final SCMCommand command,
                      final SCMConnectionManager connectionManager) {
     final DeleteContainerCommand deleteContainerCommand =
         (DeleteContainerCommand) command;
+    if (shouldRequeue(command, context, ozoneContainer)) {
+      context.requeueCommand(command);
+    }
```

Comment:
If the delete request is abandoned by datanode, it might need to wait for the next container report to resend the request.

Reply:
I think in that case we do not need the requeue functionality. Requeue itself is a complex mechanism because it will keep occurring in a loop.
Why aren't we blocking the operations on container lock instead of requeue mechanism? Is it because the handler thread is getting blocked in that case?

Reply:
Yes. Blocking is the original idea. But it might causing the thread pool of handler being blocked.

Reply:
What is the approach we are going with here then? Can we also update the docs once the PR is merged?
If we fail the operation in DN, then SCM would at least know that disk balancer is running. Later we can add a change to avoid block deletions for that specific container.